### PR TITLE
New TC for Deleting All backup objects

### PR DESCRIPTION
**Pull request by skonda-px**
**What this PR does / why we need it**:
 Adding new Test Case for Deleting all backup objects

**Which issue(s) this PR fixes** (optional)
Closes #PA-519

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by skonda-px** [comment]
Aetos DashBoard link: http://aetos.pwx.purestorage.com/resultSet/testSetID/84871

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -6009,6 +6009,194 @@ var _ = Describe("{IssueMultipleRestoresWithNamespaceAndStorageClassMapping}", f
...
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying backup cluster %s", SourceClusterName))
+		})
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Getting context")
+			for _, namespace := range bkpNamespaces {
+				backupName = fmt.Sprintf("%s-%s", BackupNamePrefix, namespace)
+				err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, []string{namespace},
+					labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:6128]: Use the pre and post rule

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -6009,6 +6009,194 @@ var _ = Describe("{IssueMultipleRestoresWithNamespaceAndStorageClassMapping}", f
...
+			log.InfoD("Creating a backup schedule policy")
+			intervalSchedulePolicyInfo := Inst().Backup.CreateIntervalSchedulePolicy(5, 15, 2)
+			intervalPolicyStatus := Inst().Backup.BackupSchedulePolicy(intervalName, uuid.New(), orgID, intervalSchedulePolicyInfo)
+			dash.VerifyFatal(intervalPolicyStatus, nil, fmt.Sprintf("Creating interval schedule policy %s", intervalName))
+		})
+		Step("Register cluster for backup", func() {
+			log.InfoD("Register cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:6117]: Handle the error here

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -6009,6 +6009,194 @@ var _ = Describe("{IssueMultipleRestoresWithNamespaceAndStorageClassMapping}", f
...
+				err = CreateRestore(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying %s backup's restore %s creation", backupName, restoreName))
+			}
+		})
+
+		Step("Delete the restores", func() {
+			log.InfoD("Delete the restores")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			DeleteRestore(restoreName, orgID, ctx)
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:6148]: Handle the error here

**Comment by skonda-px** [comment]
Aetos DashBoard Link: http://aetos.pwx.purestorage.com/resultSet/testSetID/84945/testCaseID/326641

File: tests/backup/backup_test.go [hunk 4]
```diff
@@ -6009,6 +6009,197 @@ var _ = Describe("{IssueMultipleRestoresWithNamespaceAndStorageClassMapping}", f
...
+		cloudCredUID      string
+		backupLocationUID string
+		bkpLocationName   string
+	)
+	backupLocationMap := make(map[string]string)
+	labelSelectors := make(map[string]string)
+	bkpNamespaces = make([]string, 0)
+	var namespaceMapping map[string]string
+	namespaceMapping = make(map[string]string)
+	providers := getProviders()
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:6034]: move this inside IT

**Review by lsrinivas-pure — APPROVED** [review]
Fix the comment before merging

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**